components/eamxx/scripts/jenkins/scratch_cleanup.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+"""
+Clean up old files in the scratch area for mappy.
+"""
+
+from pathlib import Path
+import re, time, shutil, sys, argparse
+
+###############################################################################
+def parse_command_line(args, description):
+###############################################################################
+    parser = argparse.ArgumentParser(
+        usage="""\n{0} [-c HOURS]
+OR
+{0} --help
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# Purge files older than 20 hours \033[0m
+    > {0} -c 20
+""".format(Path(args[0]).name),
+        description=description,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    parser.add_argument("-c", "--cutoff", type=int, default=30, help="The cutoff age for purging in hours")
+
+    parser.add_argument("-d", "--dry-run", action="store_true", help="Do a dry run, don't actually remove files")
+
+    args = parser.parse_args(args[1:])
+
+    return args
+
+###############################################################################
+def scratch_cleanup(cutoff, dry_run):
+###############################################################################
+    scratch = Path('/ascldap/users/e3sm-jenkins/acme/scratch')
```

Comment:
This script is pruning more than just scream test cases. It's pruning all e3sm-jenkins cases.

I wonder if we should just add a jenkins job that runs every 12h or so, and cleans up its scratch space. It seems more of an e3sm-jenkins duty than something that belongs to eamxx's scripts...

Reply:
E3SM has it's own test archiving system. I did a dry run and it's only cleanup up our AT stuff.